src/lib/FlightTasks/tasks/FlightTaskManualAltitude.cpp

```diff
@@ -97,7 +97,44 @@ void FlightTaskManualAltitude::_updateAltitudeLock()
 	// check if vehicle has stopped
 	const bool stopped = (MPC_HOLD_MAX_Z.get() < FLT_EPSILON || fabsf(_velocity(2)) < MPC_HOLD_MAX_Z.get());
 
-	if (MPC_ALT_MODE.get() && PX4_ISFINITE(_dist_to_bottom)) {
+	if (MPC_ALT_MODE.get() == 2) {
+		// terrain hold
+		float spd_xy = Vector2f(&_velocity(0)).length();
+		bool stick_input = Vector2f(&_velocity_setpoint(0)).length() > 0.5f * MPC_ALT_MODE_SPD.get();
```

Comment:
can we not use `_sticks` instead (https://github.com/PX4/Firmware/pull/9731/files#diff-7dfb661344d42aeefa71f4afd9b85be8R60), which would make it scaling independent.
In addition, there is already a parameter for horizontal direction that defines the speed threshold at which the vehicle is considered at rest: https://github.com/PX4/Firmware/blob/master/src/modules/mc_pos_control/mc_pos_control_params.c#L375

Reply:
Good idea. Will fix.